File: benchmark/results/ts-multilang/run-003/b-baseline/output/src/domain/identity/services/UserRepository.ts
import { User } from "../aggregates/User";

/** UserRepository domain service for the identity domain. */
export class UserRepository {
  create(input: Partial<User>): User {
    void input;
    throw new Error("not implemented");
  }

  findById(id: string): User | null {
    void id;
    throw new Error("not implemented");
  }

  list(limit: number): User[] {
    void limit;
    throw new Error("not implemented");
  }

  delete(id: string): void {
    void id;
    throw new Error("not implemented");
  }
}

File: benchmark/results/python-multilang/run-002/b-baseline/output/metaengine_demo/identity/services/user_repository.py
"""UserRepository for the identity domain."""

from typing import Optional

from metaengine_demo.identity.aggregates.user import User


class UserRepository:
    """Persistence boundary for the User aggregate."""

    def create(self, input: Optional[User]) -> User:
        """Persist a new user from a partial input."""
        raise NotImplementedError("not implemented")

    def findById(self, id: str) -> Optional[User]:
        """Load a user by its identifier."""
        raise NotImplementedError("not implemented")

    def list(self, limit: float) -> list[User]:
        """List persisted users up to the given limit."""
        raise NotImplementedError("not implemented")

    def delete(self, id: str) -> None:
        """Delete a persisted user by identifier."""
        raise NotImplementedError("not implemented")
